### DISCUSSION TITLE
Database Key Rotation

### DISCUSSION THREAD

**Discussion by pranaygp** [hasura/graphql-engine#7523] [Question] [answered]
We have a Lambda which rotates our Postgres database password every 30 days. I updated the lambda to also update our Hasura Cloud env variable during the rotation. 
That works, but, Hasura doesn't start using the new Postgres connection string automatically. After a rotation, Hasura's Data Manager shows me this: 
![image](https://user-images.githubusercontent.com/1797812/132430369-f7267c91-93a3-4eef-b5aa-83ffc3343884.png)

After I manually hit 'refresh', it grabs the latest password and the metadata is sync'd and my database works again.
![image](https://user-images.githubusercontent.com/1797812/132430905-8f92f08c-f932-473c-9326-5ab878b2e954.png)

Is there a way to programatically do this, so I can automatically resync metadata immediately after a rotation to prevent downtime?

**Comment by BenoitRanque** [comment, marked as answer]
The hasura cli has the util `hasura metadata reload`, perhaps this is what you are looking for?
https://hasura.io/docs/latest/graphql/core/hasura-cli/hasura_metadata_reload.html

**Reply by pranaygp** [reply]
Classic. I didn't look hard enough. Thanks :)
